syscall_bench_test.go

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2025 The Ebitengine Authors
+
+package purego
+
+import (
+	"fmt"
+	"testing"
+)
+
+// BenchmarkCallingMethods compares RegisterFunc, SyscallN, and Callback methods
+func BenchmarkCallingMethods(b *testing.B) {
+	testCases := []struct {
+		n     int
+		fn    any
+		fnPtr uintptr
+		args  []uintptr
+	}{
+		{1, noop1, 0, []uintptr{1}},
+		{2, noop2, 0, []uintptr{1, 2}},
+		{3, noop3, 0, []uintptr{1, 2, 3}},
+		{5, noop5, 0, []uintptr{1, 2, 3, 4, 5}},
+		{10, noop10, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+		{15, noop15, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}},
+	}
+
+	// Create callbacks once for all tests
+	for i := range testCases {
```

Comment:
~~I don't think this can be compiled with old Go versions, but is this file compiled in the CI?~~

NVM